### PR TITLE
Fix SECURITY-1025

### PR DESCRIPTION
**Pull request by lacostej**
I hope this fixes SECURITY-1025. Any way to check if there are other paths to protect?

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by MarkEWaite** [comment]
> Could anyone please replay this PR build with the updated Jenkinsfile?

Done

**Comment by lacostej** [comment]
I think I now have handled all cases that needed to be handled.

image

File: src/main/java/hudson/plugins/batch_task/BatchTaskInvoker.java [hunk 1]
```diff
@@ -129,6 +130,7 @@ public String getDisplayName() {
                 return "";
             }
 
+            @POST
```
Comment (yaroslavafenkin) [inline src/main/java/hudson/plugins/batch_task/BatchTaskInvoker.java]: This one and the other two annotations you've added in this class seem redundant to me. Annotated methods do not have any side effects, so GET is fine for these.

File: src/main/java/hudson/plugins/batch_task/BatchTask.java [hunk 2]
```diff
@@ -264,6 +265,7 @@ public Object getDynamic(String token, StaplerRequest req, StaplerResponse rsp)
     /**
      * Schedules the execution
      */
+    @POST
```
Comment (yaroslavafenkin) [inline src/main/java/hudson/plugins/batch_task/BatchTask.java:268]: With addition of these annotations UI buttons seem to be broken.
I've create a dummy freestyle job and added 2 batch tasks to it. Then when I go to `JENKINS/job/JOB_NAME/batchTasks/task/TASK_NAME/` and click "Build Now" I get a 404. Same with "Delete Task".

I suspect that culprits are https://github.com/jenkinsci/batch-task-plugin/blob/master/src/main/resources/hudson/plugins/batch_task/BatchTask/sidepanel.jelly#L18 and https://github.com/jenkinsci/batch-task-plugin/blob/master/src/main/resources/hudson/plugins/batch_task/BatchTask/delete.jelly#L9C11-L9C23

For `l:task` there exists a `post` attribute, so the fix should be straightforward.

File: src/test/java/hudson/plugins/batch_task/TestHelper.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Performs an HTTP POST request to the relative url.
+     *
+     * @param webClient the client
+     * @param relative the url relative to the context path
+     * @param expectedContentType if expecting specific content type or null if not
+     * @param expectedStatus if expecting a failing http status code or null if not
+     * @throws IOException if so
+     */
+    public static HtmlPage post(JenkinsRule.WebClient webClient, String relative,
```
Comment (yaroslavafenkin) [inline src/test/java/hudson/plugins/batch_task/TestHelper.java]: FYI utility methods exist in `jenkins-test-harness` that allow you to send post requests already.
See https://github.com/jenkinsci/jenkins/blob/9012a5b35aca298e4cfa23c6ca95242690c232c3/test/src/test/java/hudson/model/ViewTest.java#L264-L267 for example.

File: src/main/java/hudson/plugins/batch_task/BatchTaskInvoker.java (hunk 1 above)
Comment (lacostej) [reply src/main/java/hudson/plugins/batch_task/BatchTaskInvoker.java]: Thanks for the review. Removed.

File: src/test/java/hudson/plugins/batch_task/TestHelper.java (hunk 3 above)
Comment (lacostej) [reply src/test/java/hudson/plugins/batch_task/TestHelper.java]: Thanks, I adjusted my helper method.

File: src/main/java/hudson/plugins/batch_task/BatchTask.java (hunk 2 above)
Comment (lacostej) [reply src/main/java/hudson/plugins/batch_task/BatchTask.java:268]: perfect. Note that I removed the delete.jelly as it does no seem to be used.

I am still able to delete tasks without it.

File: src/main/java/hudson/plugins/batch_task/BatchRun.java [hunk 4]
```diff
@@ -326,6 +327,7 @@ public String getUrlName() {
     /**
      * Handles incremental log output.
      */
+    @POST
```
Comment (yaroslavafenkin) [inline src/main/java/hudson/plugins/batch_task/BatchRun.java]: This is not really needed, no CSRF here.

**Review by yaroslavafenkin — APPROVED** [review]
Works well in local testing.

**Pull request merged**